catboost/libs/metrics/metric.cpp

```diff
@@ -544,6 +545,84 @@ TVector<TParamSet> TMultiRMSEMetric::ValidParamSets() {
     return {TParamSet{{TParamInfo{"use_weights", false, true}}, ""}};
 };
 
+/* MultiRMSEWithMissingValues */
```

Comment:
let's add check for NaNs for all metrics in CheckPreprocessedTarget https://github.com/catboost/catboost/blob/74c9ca7e62d01236199e1bf44decd05586df6624/catboost/libs/metrics/metric.cpp#L6447 for fast fail if user tries to calc something with NaNs except of MultiRMSEWithMissingValues and UserDefined metrics

Reply:
ok. New proposal in master at the end of CheckPreprocessedTarget.

Reply:
I wrote some implementation there... the current CheckProcessedTarget works very well and break when Nan are found. Now, the new version propose the possibility to skip Nans if MultiRMSEWithMissingValues or UserDefined metrics. It seems to be fine.

Reply:
Actually we have 2 functions with name `CheckPreprocessedTarget` - the first one is in [data_providers.cpp](https://github.com/catboost/catboost/pull/1680/files#diff-193809f26322d2a182596e906a6cfc5ed809066fb14c5fc68ba57cb96eb22449R100) and the second one is in this file - `metric.cpp`. I mean to add checks to the second one, because it is called for each metric, not for loss function only. Now if the user fit CatBoost with NaNs and `loss_function="MultiRMSEWithMissingValues"` and set `eval_metric="MultiRMSE"` then he get NaNs. It will be better to get error message instead of NaNs.